src/connector/src/parser/postgres.rs

```diff
@@ -227,7 +233,37 @@ pub fn postgres_row_to_owned_row(row: tokio_postgres::Row, schema: &Schema) -> O
                             handle_list_data_type!(row, i, name, NaiveDate, builder, Date);
                         }
                         DataType::Varchar => {
-                            handle_list_data_type!(row, i, name, String, builder);
+                            match *row.columns()[i].type_() {
+                                // Since we don't support UUID natively, adapt it to a VARCHAR column
+                                Type::UUID => {
+                                    let res = row.try_get::<_, Option<uuid::Uuid>>(i);
+                                    match res {
+                                        Ok(val) => {
+                                            if let Some(v) = val {
+                                                builder
+                                                    .append(Some(ScalarImpl::from(v.to_string())));
+                                            }
+                                        }
+                                        Err(err) => {
+                                            if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
+                                                tracing::error!(
+                                                    suppressed_count,
+                                                    column = name,
+                                                    error = %err.as_report(),
+                                                    "parse uuid column failed",
+                                                );
+                                            }
+                                        }
+                                    };
+                                }
+                                // support converting NUMERIC to VARCHAR implicitly
+                                Type::NUMERIC => {
+                                    let _ = postgres_decimal_to_varchar(&row, i, name, None);
```

Comment:
Return value is discarded and no array builder passed in. Is this the desired behavior?

Reply:
refactored.